modules/charset/escape.go

```diff
@@ -0,0 +1,229 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package charset
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+	"unicode"
+	"unicode/utf8"
+
+	"golang.org/x/text/unicode/bidi"
+)
+
+// EscapeStatus represents the findings of the unicode escaper
+type EscapeStatus struct {
+	Escaped      bool
+	HasError     bool
+	HasBadRunes  bool
+	HasControls  bool
+	HasSpaces    bool
+	HasMarks     bool
+	HasBIDI      bool
+	BadBIDI      bool
+	HasRTLScript bool
+	HasLTRScript bool
+}
+
+// Or combines two EscapeStatus structs into one representing the conjunction of the two
+func (status EscapeStatus) Or(other EscapeStatus) EscapeStatus {
+	status.Escaped = status.Escaped || other.Escaped
+	status.HasError = status.HasError || other.HasError
+	status.HasBadRunes = status.HasBadRunes || other.HasBadRunes
+	status.HasControls = status.HasControls || other.HasControls
+	status.HasSpaces = status.HasSpaces || other.HasSpaces
+	status.HasMarks = status.HasMarks || other.HasMarks
+	status.HasBIDI = status.HasBIDI || other.HasBIDI
+	status.BadBIDI = status.BadBIDI || other.BadBIDI
+	status.HasRTLScript = status.HasRTLScript || other.HasRTLScript
+	status.HasLTRScript = status.HasLTRScript || other.HasLTRScript
```

Comment:
```suggestion
	status.Escaped |= other.Escaped
	status.HasError |= other.HasError
	status.HasBadRunes |= other.HasBadRunes
	status.HasControls |= other.HasControls
	status.HasSpaces |= other.HasSpaces
	status.HasMarks |= other.HasMarks
	status.HasBIDI |= other.HasBIDI
	status.BadBIDI |= other.BadBIDI
	status.HasRTLScript |= other.HasRTLScript
	status.HasLTRScript |= other.HasLTRScript
```

Reply:
These operations can be simplified.

Reply:
I am sure I tried that but it wouldn't compile.

Reply:
Woops they are `bool` not `int` 🤦🏽